src/pipeline.cc

```diff
@@ -989,6 +1034,9 @@ class PipelineWorker : public AsyncWorker {
     if (baton->overlayBufferInLength > 0) {
       GetFromPersistent("overlayBufferIn");
     }
+    if (baton->booleanBufferInLength > 0) {
```

Comment:
How about a single call to `GetFromPersistent` within a for loop over `saveBuffers`?